assets/src/editor/events/ticket-assignments-manager/ticket-assignments-manager-modal.js

```diff
@@ -865,67 +661,95 @@ export default compose( [
 				'event_espresso'
 			),
 		};
+		const { getRelatedEntities } = select( 'eventespresso/core' );
+
+		// initial setup based on incoming entity
 		if ( isModelEntityOfModel( dateEntity, 'datetime' ) ) {
 			dtmProps.entities = [ dateEntity ];
 		} else if ( isModelEntityOfModel( ticketEntity, 'ticket' ) ) {
 			dtmProps.entities = sortDateEntitiesList( dateEntities );
 			dtmProps.ticketEntities = [ ticketEntity ];
-		} else if ( Array.isArray( dateEntities ) && Array.isArray( ticketEntities ) ) {
+		} else {
 			dtmProps.entities = sortDateEntitiesList( dateEntities );
 		}
-		const { getRelatedEntities } = select( 'eventespresso/core' );
-		const ticketEntitiesByDateIds = {};
+
+		// need to setup the assignmentCounts for all the tickets and all the dates!
 		dtmProps.entities.forEach( ( date ) => {
-			if ( isModelEntityOfModel( date, 'datetime' ) ) {
+			if ( typeof assignmentCounts.current.dates[ date.id ] === 'undefined' ) {
 				const relatedTickets = getRelatedEntities( date, 'ticket' );
-				ticketEntitiesByDateIds[ date.id ] = uniq( relatedTickets );
+				dtmProps.ticketEntitiesByDateIds[ date.id ] = relatedTickets;
+				assignmentCounts.current.dates[ date.id ] = relatedTickets.length || 0;
 			}
 		} );
-		return {
-			...dtmProps,
-			ticketEntitiesByDateIds,
-		};
+		dtmProps.ticketEntities.forEach( ( ticket ) => {
+			if ( typeof assignmentCounts.current.tickets[ ticket.id ] === 'undefined' ) {
+				assignmentCounts.current.tickets[ ticket.id ] = getRelatedEntities(
+					ticket,
+					'datetime'
+				).length || 0;
+				// no need to set dtmProps.ticketEntitiesByDateIds here as
+				// those will already have been setup for all dates.
+			}
+		} );
+		return dtmProps;
```

Comment:
So part of the reason the reducer were looping over assignedState _and_ assignedCounts in the original iteration of the custom hook is because it was possible for the passed in `assignedCounts` to not have ticket or date ids in it’s collection that existed in `assignedState` due to how `assignedCounts` was built in the composition of `TicketAssignmentsManagerModal`.  Hence, the refactoring in here.  Now, there is always a count set for the tickets and dates being used in the session.